lighthouse-core/config/config-plugin.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @param {unknown} arr
+ * @return {arr is Array<Record<string, unknown>>}
+ */
+function isArrayOfUnknownObjects(arr) {
+  return Array.isArray(arr) && arr.every(isObjectOfUnknownProperties);
+}
+
+/**
+ * @param {unknown} val
+ * @return {val is Record<string, unknown>}
+ */
+function isObjectOfUnknownProperties(val) {
+  return typeof val === 'object' && val !== null;
+}
+
+/**
+ * A set of methods for extracting and validating a Lighthouse plugin config.
+ */
+class ConfigPlugin {
+  /**
+   * Extract and validate the list of AuditDefns added by the plugin (or undefined
+   * if no additional audits are being added by the plugin).
+   * @param {unknown} auditsJson
+   * @param {string} fullPluginName
+   * @return {Array<{path: string}>|undefined}
+   */
+  static _parseAuditsList(auditsJson, fullPluginName) {
+    // Plugin audits aren't required (relying on LH default audits) so fall back to [].
+    if (auditsJson === undefined) {
+      return undefined;
+    } else if (!isArrayOfUnknownObjects(auditsJson)) {
```

Comment:
should we add the TODO of supporting strings then?

already tracked by an issue?

Reply:
> should we add the TODO of supporting strings then? already tracked by an issue?

I'll start collecting things for a follow up issue